src/porepy/grids/mortar_grid.py

```diff
@@ -480,8 +436,10 @@ def master_to_mortar_avg(self, nd: int = 1) -> sps.spmatrix:
                 Size: g_master.num_faces x mortar_grid.num_cells.
 
         """
-        scaled_mat = self._row_sum_scaling_matrix(self._master_to_mortar_int)
```

Comment:
See comments in `_row_sum_scaling`: Using a diagonal matrix actually takes more time, and in certain cases it can be a substantial factor. Please revert.

Reply:
Same comment for the slave_to_mortar

Reply:
Same as in compute_geometry(). This is a mistake when merging the latest changes from develop into the branch. git kept the wrong version for some reason (that is probably my fault). Can you have a check and see if all the functions in the MortarGrid class are correct ? In this pull request it should really only be the constructor that changed